kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaServiceStatusFactory.java

```diff
@@ -34,36 +37,50 @@ private KafkaService serviceStatusPatch(KafkaService observedIngress,
         if (!checksum.equals(MetadataChecksumGenerator.NO_CHECKSUM_SPECIFIED)) {
             Annotations.annotateWithReferentChecksum(metadataBuilder, checksum);
         }
-        return metadataBuilder
+
+        KafkaServiceBuilder service = metadataBuilder
                 .endMetadata()
                 .withNewStatus()
                     .withObservedGeneration(ResourcesUtil.generation(observedIngress))
                     .withConditions(ResourceState.newConditions(Optional.ofNullable(observedIngress.getStatus()).map(KafkaServiceStatus::getConditions).orElse(List.of()), ResourceState.of(condition)))
-                .endStatus()
-                .build();
+                .endStatus();
+
+        if (bootstrapServers != null && !bootstrapServers.isEmpty()) {
+            service.editStatus().withBootstrapServers(bootstrapServers).endStatus();
```

Comment:
I would collapse the `if` into the `withNewStatus` 

```java
                .withNewStatus()
                    .withObservedGeneration(ResourcesUtil.generation(observedIngress))
                    .withConditions(ResourceState.newConditions(Optional.ofNullable(observedIngress.getStatus()).map(KafkaServiceStatus::getConditions).orElse(List.of()), ResourceState.of(condition)))
.withBootstrapServers(Optional.ofNullable(bootstrapServers).filter(Predicate.not(String::isEmpty)).orElse(null))
endStatus();

Reply:
Since we have decided to remove check for `!bootstrapServers.isEmpty()` should this condition shorten to just `withBootstrapServers(Optional.ofNullable(bootstrapServers).orElse(null)` which I think would be equivalent to ``withBootstrapServers(bootstrapServers)`